### PR TITLE
Add NGPV rule to assert repository metadata is set in all packages

### PR DESCRIPTION
**Pull request by natemcmaster**
Add tests to check that repository commit is set in our nupkgs.

Also, add commit to .Sources packages

### CODE REVIEW

File: modules/NuGetPackageVerifier/console/Rules/PackageSigningrule.cs [hunk 1]
```diff
@@ -31,9 +32,17 @@ public PackageSigningRule(string exePath)
...
-            // KoreBuild/modules/NuGetPackageVerifier/
-            return Path.Combine(AppContext.BaseDirectory, "..", "..", "nuget.exe");
+            var searchPaths = new[]
+            {
+                // KoreBuild/nuget.exe
+                // KoreBuild/modules/NuGetPackageVerifier/
+                Path.Combine(AppContext.BaseDirectory, "..", "..", "nuget.exe"),
+
+                // obj/nuget.exe in local build root
+                Path.Combine(AppContext.BaseDirectory, "..", "..", "..", "..", "..", "..", "obj", "nuget.exe"),
```
Comment (pranavkm) [inline modules/NuGetPackageVerifier/console/Rules/PackageSigningrule.cs]: lol

File: modules/NuGetPackageVerifier/console/Rules/PackageSigningrule.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
```
Comment (pranavkm) [inline modules/NuGetPackageVerifier/console/Rules/PackageSigningrule.cs]: Could you also fixup the file name?

File: test/NuGetPackageVerifier.Tests/Rules/PackageRepoMetadataRuleTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
```
Comment (pranavkm) [inline test/NuGetPackageVerifier.Tests/Rules/PackageRepoMetadataRuleTests.cs]: Remove & sort usings?

Comment (natemcmaster) [reply test/NuGetPackageVerifier.Tests/Rules/PackageRepoMetadataRuleTests.cs]: They're all used :)

**Pull request merged**